reward_index/src/indexer.rs

```diff
@@ -150,6 +152,19 @@ impl Indexer {
                         },
                         r.discovery_location_amount,
                     )),
+                    Some(MobileReward::ServiceProviderReward(r)) => {
+                        if let Some(sp) = ServiceProvider::from_i32(r.service_provider_id) {
+                            Ok((
+                                RewardKey {
+                                    key: sp.as_str_name().to_string(),
+                                    reward_type: RewardType::MobileSubscriber,
```

Comment:
is this the right reward type for a service provider?

Reply:
it most def is not, addressed here: 3f5924313ce8efe32a580e1981be7f4018355ee8